code/script.js

```diff
@@ -218,108 +256,233 @@ const generateBoard = () => {
+      customAlert(`Yes, the person is a ${value}! Keep all people with ${value} appearance.`);
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      customAlert(`No, the person is not a ${value}! Remove all people that have a ${value} appearance.`);
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
     }
   }
+  generateBoard();
+};
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+// My Initial code that was showing a default browser confirmation and calling checkMyGuess if confirmed
+// const guess = (personToConfirm) => {
+//   let confirmation = confirm("You want to make a guess?");
+//   if (confirmation) {
+//     checkMyGuess(personToConfirm);
+//   }
+// };
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
-
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// Connecting a customized confirmation popup https://getbootstrap.com/docs/4.0/components/modal/
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  $("#myModal").modal("show"); /* Thats a bootstrap modal component - called with use of jQuery*/
+  selectedPerson = personToConfirm;
+};
 
-// If you confirm, this function is invoked
+// If user confirms that he/she wants to make a guess, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (personToCheck === secret.name) {
+    showWinScreen();
+    /* playes audio for a winner on the Desktop version*/
+    audio.play();
+    /* to control the audio volume*/
+    audio.volume = 0.2;
+  } else {
+    showLoseScreen();
+  }
+};
+
+// This function is drawing the winning screen provides an info on number of attempts being made, showing
+// the picture of correct character
+const showWinScreen = () => {
+  winOrLoseText.innerText = `You Won in ${attempts} attempts!`;
```

Comment:
I like that you didn't just display this immediately but used it in the win page